packages/firestore/src/remote/rpc_error.ts

```diff
@@ -243,9 +248,9 @@ export function mapCodeFromHttpStatus(status: number): Code {
       return Code.OK;
 
     case 400: // Bad Request
-      return Code.INVALID_ARGUMENT;
+      return Code.FAILED_PRECONDITION;
```

Comment:
This doesn't seem right. A 400 status most typically indicates that the client has sent a bad request, not that the request is valid but can't be performed now. `INVALID_ARGUMENT` is the primary status in https://cloud.google.com/apis/design/errors#handling_errors.

Reply:
So we currently send this request to write a non-existing document in a transaction:

```
  writes: [
    {
      update: {
        name: 'projects/test-emulator/databases/(default)/documents/test-collection/noHB7gQj6Tp98raxkpxU',
        fields: { counter: { integerValue: '1' } }
      },
      currentDocument: { updateTime: '1970-01-01T00:00:00.000000000Z' }
    }
  ]
```

This `updateTime: epoch` trick/hack seems to do the same thing as `{ currentDocument: { exists: false }}` but it returns a 400 (against Prod and the emulator). If we return INVALID_ARGUMENT here, the transaction is not retried.

One thing to note is that `mapCodeFromHttpStatus` was actually never used before, so at least this change should not affect any existing callsites.